shell/browser/api/event_emitter.h

```diff
@@ -107,15 +107,15 @@ class EventEmitter : public Wrappable<T> {
                      v8::Local<v8::Object> event,
                      Args&&... args) {
     DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
-    v8::Locker locker(isolate());
-    v8::HandleScope handle_scope(isolate());
-    EmitEvent(isolate(), GetWrapper(), name, event,
-              std::forward<Args>(args)...);
-    auto context = isolate()->GetCurrentContext();
+    auto* isolate = this->isolate();
```

Comment:
what motivated this change?

Reply:
`this` is destroyed by `EmitEvent`. so `this->isolate()` crashes.

Reply:
```suggestion
    // It's possible that |this| will be deleted by EmitEvent, so save anything we need
    // from |this| before calling EmitEvent.
    auto* isolate = this->isolate();
```